pandas/core/ops.py

```diff
@@ -754,7 +754,9 @@ def wrapper(self, other, axis=None):
         elif isinstance(other, pd.DataFrame):  # pragma: no cover
             return NotImplemented
         elif isinstance(other, (np.ndarray, pd.Index)):
-            if len(self) != len(other):
+            # do not check length of zerorank array
+            if not lib.isscalar(lib.item_from_zerodim(other)) and \
+               len(self) != len(other):
                 raise ValueError('Lengths must match to compare')
```

Comment:
either one are fine really. I guess in this case parens might be nicer. I'll fix on the merge